### PR TITLE
[AArch64] Add assembly/disassembly for BFMOP4{A,S} (widening) instructions

### PR DESCRIPTION
**Pull request by momchil-velikov**
The new instructions are described in
https://developer.arm.com/documentation/ddi0602/2024-09/SME-Instructions

### CODE REVIEW

**Review by CarolineConcatto — APPROVED** [review]

File: llvm/lib/Target/AArch64/SMEInstrFormats.td [hunk 1]
```diff
@@ -5126,3 +5126,40 @@ class sme2_luti4_vector_vg4_strided<bits<2> sz, bits<2> op, string mnemonic>
...
+  let Inst{4} = S;
+  let Inst{3-2} = 0b00;
+  let Inst{1-0} = ZAda;
+
+  let Constraints = "$ZAda = $_ZAda";
+}
+
+multiclass sme2_bfmop4as_widening<bit S, string mnemonic> {
+  // Single vectors
+  def _ZZ_S : sme2_bf16_fp32_quarter_tile_outer_product<0, 0, S, mnemonic, ZPR16Mul2_Lo, ZPR16Mul2_Hi>;
```
Comment (SpencerAbson) [inline llvm/lib/Target/AArch64/SMEInstrFormats.td:5155]: Just having a skim over these patches - IIUC, the names here should start with `M` to denote their use of `Za` as an operand.

Comment (CarolineConcatto) [reply llvm/lib/Target/AArch64/SMEInstrFormats.td:5155]: Yes, I believe Spencer is correct. I had forgotten about that too:

// M = SME array register (ZA)
// P = Predicate register
// C = Predicate-as-counter register
// I = immediate
// Z = SVE vector register
// T = ZT0 register

Comment (momchil-velikov) [reply llvm/lib/Target/AArch64/SMEInstrFormats.td]: Done.

**Review by SpencerAbson — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**